### ISSUE TITLE
Changing Text Source while streaming hangs OBS

### ISSUE THREAD

**Issue by mpek** [obsproject/obs-studio#9037]
### Operating System Info

Windows 11

### OBS Studio Version

29.1.2

### OBS Studio Log URL

https://obsproject.com/logs/oAUBe9eDovaQ2EZN

### Expected Behavior

Uploaded OBS Studio Log URL is the previous log!!! I uploaded the specific one here: 
[2023-06-06 15-04-50.txt](https://github.com/obsproject/obs-studio/files/11671077/2023-06-06.15-04-50.txt)

Don't let OBS hang/block its usability, let it encode video correctly, after changing and still streaming

### Current Behavior

Textsource changed while streaming, suddenly OBS hangs and does not react to any user interaction.
Stream gets wrong encoded video, changes between grey and normal and mixed video (mixed = grey areas and color areas in the video). 

In that scene in question are these sources: 

Two Text (GDI+) Sources
One Text (GDI+) Slide Show Source 
Capture USB Video Source (1080)
VLC Video Source (playing audio files)

I cannot upload log through OBS, because there are already two new logs. You'll get it attached here. 

[2023-06-06 15-04-50.txt](https://github.com/obsproject/obs-studio/files/11671077/2023-06-06.15-04-50.txt)

### Steps to Reproduce

1. Stream 8 hours to three different targets
2. Change the scene to show replays (VLC Video Source)
3. Change to studio modus
4. Change one of the current scene's text source: Change font, edit transform
5. Hit Transition (between Preview and Program
6. Hangs
...

### Anything else we should know?

In that scene in question are

Two Text (GDI+) Sources
One Text (GDI+) Slide Show Source 
Capture USB Video Source Full HD (EZCOO USB Box)
VLC Video Source (playing audio files)

Text Slide Show is in alpha and it cannot be ruled out that it has something to do with it. 
However, an OBS own text source was being edited at the time of hanging/blocking OBS. 
https://obsproject.com/forum/threads/obs-text-slideshow.144903/

**Comment by prgmitchell** [comment]
These reproduction steps are not very simple and take a long time so I think it may be tough for someone to try. I would try disabling Hardware Accelerated GPU Scheduling in your Windows settings as it it the most common cause of hangs in OBS and you currently have it enabled according to the log.

Where to disable on Windows 11:
Settings → System → Display → Graphics → Change default graphics settings
From there, set "Hardware-accelerated GPU scheduling" to OFF, then reboot your computer.

**Comment by notr1ch** [comment]
You have a huge amount of 3rd party plugins installed which are more likely responsible for the issue than the core OBS plugins. Unless you can narrow down a test case that shows that this problem exists in OBS without any 3rd party plugins, this report can't really be actioned on as it's not practical to reproduce your setup.

If the problem isn't resolved by disabling HAGS, sharing a [dump file](https://obsproject.com/forum/resources/how-to-create-a-dmp-file-for-debugging.598/) may help pinpoint which plugin is responsible.

**Comment by mpek** [comment]
@prgmitchell 
Thank you for the reference to HAGS. Will be turned off and see when I stream again for that long and this situation can be replicated. This problem does not occur when freshly started computer and OBS. 

@notr1ch 
I know what you mean, it is better to reduce additional code to the necessary, but currently trying streaming with every plugin that sounds interesting and could benefit a stream. Surely, this could cause problems, but if we don't try, what's the point they exist? :) 
I'll remember the dump and do if OBS freezes again. Who should the link to that dump be sent to privately?

**Comment by RytoEX** [comment]
> @prgmitchell Thank you for the reference to HAGS. Will be turned off and see when I stream again for that long and this situation can be replicated. This problem does not occur when freshly started computer and OBS.
> 
> @notr1ch I know what you mean, it is better to reduce additional code to the necessary, but currently trying streaming with every plugin that sounds interesting and could benefit a stream. Surely, this could cause problems, but if we don't try, what's the point they exist? :) I'll remember the dump and do if OBS freezes again. Who should the link to that dump be sent to privately?

The point of trying without plugins is to see if the problem occurs without any plugins. If it doesn't, then it's probably a plugin causing the crash. Then you add plugins back in one at a time until you reproduce the crash, then you try with just that plugin. Repeat until you can reproduce the crash with just one plugin installed and then report the crash to the plugin developer. That is the point of @notr1ch's suggestion. It's a matter of determining if it's something _we_ need to fix, or if it's something that needs to be reported elsewhere for someone else to fix.

**Comment by gxalpha** [comment]
@mpek have you tried without plugins as suggested above?

**Comment by Fenrirthviti** [comment]
Without eliminating third-party plugins (which we do not develop, or maintain, and as such would not be able to fix) it is hard to focus on where or what the issue might be. Testing without plugins is one of the first and most basic troubleshooting steps, as it reduces the complexity of the issue by several orders of magnitude.

**Comment by mpek** [comment]
Yeah, that would be best, but for now the first step is to reduce plugins step by step to be able to keep the stream going. When this is done but that failure happens again, we'll get back to that dump.
